File: C language/4.looping/12amstrongnumber.c
#include <stdio.h>
#include <math.h>

int main()
{
    int num, len = 0, i, temp, sum = 0;

    printf("Please enter an integer: "); // Taking the user input
    scanf("%d", &num);
    for (i = num; i > 0; i = i / 10)
        len++;
    temp = num;
    while (num > 0)
    {
        sum = sum + pow(num % 10, len);
        num = num / 10;
    }
    if (temp == sum)
    {
        printf("amstrong number");
    }
    else
    {
        printf("not an amstrong number");
    }
}

File: Python/6.Loops/12.amstrongnumber.py
a=input("enter number: ")
n=int(a)
b=len(a)
t=n
s=0

while n>0:
    r=n%10
    c=pow(r,b)
    n=n//10
    s=s+c
    
if s==t:
    print("amstrong number")
else:
    print("not amstrong number")
    
# 153=1^3+5^3+3^3